src/main/java/org/zalando/nakadi/repository/zookeeper/ZookeeperProxy.java

```diff
@@ -0,0 +1,121 @@
+        } catch (final KeeperException e) {
+            LOG.trace("Error creating locks", e);
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    @Override
+    public void lockAcquire(final String path, final long timeoutMs) throws RuntimeException {
+        int requestCounter = 0;
+        final byte[] sessionId = String.valueOf(zookeeper.getSessionId()).getBytes();
+        final long finishAt = System.currentTimeMillis() + timeoutMs;
+        while (finishAt > System.currentTimeMillis()) {
+            requestCounter++;
+            try {
+                zookeeper.create(path, sessionId, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.EPHEMERAL);
+                return;
+            } catch (final Exception e) {
+                if (e instanceof KeeperException) {
+                    LOG.error("Error while creating lock node, will check session id", e);
+                    try {
+                        final byte[] storedSessionId = zookeeper.getData(path, false, null);
+                        if (Arrays.equals(sessionId, storedSessionId)) {
+                            return;
+                        }
+                    } catch (final Exception e2) {
+                        processException(e2);
+                    }
+                }
+                processException(e);
+            }
+
+            retryDelay(requestCounter);
+        }
+
+        throw new RuntimeException("Timeout reached acquiring lock");
+    }
+
+    @Override
+    public void lockRelease(final String path) throws RuntimeException {
+        int requestCounter = 0;
+        final byte[] sessionId = String.valueOf(zookeeper.getSessionId()).getBytes();
+        while (requestCounter < RETRY_MAX_COUNT) {
+            requestCounter++;
+            try {
+                final byte[] storedSessionId = zookeeper.getData(path, false, null);
+                if (Arrays.equals(sessionId, storedSessionId)) {
+                    try {
+                        zookeeper.delete(path, -1);
+                        return;
+                    } catch (final Exception e) {
+                        processException(e);
+                    }
+                } else {
+                    throw new RuntimeException("Can not release the lock which was not acquired");
+                }
+            } catch (final Exception e) {
+                processException(e);
+            }
+
+            retryDelay(requestCounter);
+        }
+        LOG.error("Failed to release lock for {}", path);
+    }
+
+    @Override
+    public void close() {
+        try {
+            zookeeper.close();
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOG.error("Interrupted while closing zookeeper client", e);
+        }
+    }
+
+
+    private void retryDelay(final int requestCounter) {
```

Comment:
Killer method!
In case of interruption - hang up for specified amount of seconds constantly knocking to zk